DOCUMENTATION.md

```diff
@@ -1,810 +1,780 @@
+  name: 'John Smith', # field from template
+  application_date: '2018-01-01' # field from template
+},
+```
 
-letter.id           # => uuid for the notification
-letter.reference    # => Reference string you supplied in the request
-letter.content      # => Hash containing body => the body of the letter sent to the recipient, with placeholders replaced
-                    #                    subject => the main heading of the letter
-letter.template     # => Hash containing id => id of the template
-                    #                    version => version of the template
-                    #                    uri => url of the template
-letter.uri          # => URL of the notification
+#### personalisation (optional)
+
+The following parameters in the letter recipient's address are optional:
+
+```ruby
+personalisation: {
+  address_line_3: 'Richmond',  # optional address field
+  address_line_4: 'London', # optional address field
+  address_line_5: 'Middlesex', # optional address field
+  address_line_6: 'UK', # optional address field
+},
 ```
 
-|`error.code`|`error.message`|
-|:---|:---|
-|`429`|`[{`<br>`"error": "RateLimitError",`<br>`"message": "Exceeded rate limit for key type TEAM of 10 requests per 10 seconds"`<br>`}]`|
-|`429`|`[{`<br>`"error": "TooManyRequestsError",`<br>`"message": "Exceeded send limits (50) for today"`<br>`}]`|
-|`400`|`[{`<br>`"error": "BadRequestError",`<br>`"message": "Can"t send to this recipient using a team-only API key"`<br>`]}`|
-|`400`|`[{`<br>`"error": "BadRequestError",`<br>`"message": "Can"t send to this recipient when service is in trial mode - see https://www.notifications.service.gov.uk/trial-mode"`<br>`}]`|
-|`400`|`[{`<br>`"error": "ValidationError",`<br>`"message": "personalisation address_line_1 is a required property"`<br>`}]`|
+#### reference (optional)
 
-</details>
+A unique identifier. This reference identifies a single unique notification or a batch of notifications.
 
+```ruby
+reference: 'your_reference_string'
+```
 
-#### Arguments
+### Response
 
-<details>
-<summary>
-Click here to expand for more information.
-</summary>
+If the request to the client is successful, the client will return a `Notifications::Client:ResponseNotification` object. This object is named `letterresponse` in the method example.
 
-#### `template_id`
-Find by clicking **API info** for the template you want to send.
+You can then call different methods on this object to return the requested information:
 
-#### `reference`
-An optional identifier you generate. The `reference` can be used as a unique reference for the notification. Because Notify does not require this reference to be unique you could also use this reference to identify a batch or group of notifications.
+|Method|Information|Type|
+|:---|:---|:---|
+|`letterresponse.id`|Notification UUID|String|
+|`letterresponse.reference`|`reference` argument|String|
+|`letterresponse.content`|- Letter body<br>- Letter subject or main heading|Hash|
```

Comment:
The two items in the hash are called `body` and `subject` (in case you decide to include this info).